### PR TITLE
support configuration commands

### PR DESCRIPTION
**Pull request by tao-sj**
support configuration commands like (interface, vlan, etc)

### CODE REVIEW

**Comment by tao-sj** [comment]
thanks for your comments, I've created two more PR for 
1. Refactoring JSON RPC interface (e.g. JSONRPCResponse, etc.)
2. Handling of multiple commands in the single JSONRPCRequest (that's where you go from string to []string

Once they are merged, I'll update this PR.

**Comment by greenpau** [comment]
> Once they are merged, I'll update this PR.

@tao-sj , this one is ready for you.

**Comment by tao-sj** [comment]
```$ make test
PASS: golint
=== RUN   TestParseShowBgpSummaryOutput
--- PASS: TestParseShowBgpSummaryOutput (0.00s)
    bgp_test.go:72: PASS: Test 0: input 'show.ip.bgp.summary.vrf.all.1', expected to pass, passed
=== RUN   TestClient
--- PASS: TestClient (0.03s)
    client_test.go:99: server: received command: show version
    client_test.go:172: client: Hostname: switch
    client_test.go:173: client: Processor Board ID: 92OB28E1IZ8
    client_test.go:174: client: Kickstart Image Version: 9.2(2)
    client_test.go:175: client: Uptime: 891
    client_test.go:176: client: took 1.208678ms
    client_test.go:99: server: received command: show interface
    client_test.go:183: client: Interfaces: 153
    client_test.go:184: client: took 17.751467ms
    client_test.go:99: server: received command: show vlan
    client_test.go:191: client: VLANs: 2
    client_test.go:192: client: took 1.042729ms
    client_test.go:99: server: received command: show system resources
    client_test.go:199: client: CPUs: 4
    client_test.go:200: client: Processes: 678
    client_test.go:201: client: took 950.5µs
    client_test.go:99: server: received command: show environment
    client_test.go:208: client: Fans: 6
    client_test.go:209: client: Power Supplies: 2
    client_test.go:210: client: Sensors: 5
    client_test.go:211: client: took 820.358µs
    client_test.go:133: server: received command: show running-config
    client_test.go:218: client: Config output size (bytes): 103
    client_test.go:219: client: took 522.069µs
    client_test.go:133: server: received command: show ip bgp summary vrf all
    client_test.go:226: client: BGP summary output size (bytes): 122
    client_test.go:227: client: took 379.871µs
    client_test.go:99: server: received command: show interface transceiver details
    client_test.go:234: client: Transceivers: 2
    client_test.go:99: server: received command: show clock
    client_test.go:118: server: received commands: [interface e1/1 shutdown]
    client_test.go:272: client: took 2.311836ms
=== RUN   TestParseShowRunningConfigurationOutput
--- PASS: TestParseShowRunningConfigurationOutput (0.00s)
    configuration_test.go:72: PASS: Test 0: input 'show.running.config.1', expected to pass, passed
=== RUN   TestParseShowInterfaceJsonOutput
--- PASS: TestParseShowInterfaceJsonOutput (0.03s)
    interface_test.go:90: PASS: Test 0: input 'show.interfaces.1', expected to pass, passed
    interface_test.go:90: PASS: Test 1: input 'show.interfaces.2', expected to pass, passed
    interface_test.go:90: PASS: Test 2: input 'show.interfaces.3', expected to pass, passed
    interface_test.go:90: PASS: Test 3: input 'show.interfaces.4', expected to pass, passed
=== RUN   TestParseShowInterfaceEthernet
--- PASS: TestParseShowInterfaceEthernet (0.00s)
    interface_test.go:113: PASS: ok
=== RUN   TestParseShowInterfaceSvi
--- PASS: TestParseShowInterfaceSvi (0.00s)
    interface_test.go:131: PASS: ok
=== RUN   TestParseShowInterfaceMgmt
--- PASS: TestParseShowInterfaceMgmt (0.00s)
    interface_test.go:149: PASS: ok
=== RUN   TestParseShowVersionJsonOutput
--- PASS: TestParseShowVersionJsonOutput (0.00s)
    sysinfo_test.go:91: PASS: Test 0: input 'show.version.1', expected to pass, passed
    sysinfo_test.go:91: PASS: Test 1: input 'show.version.2', expected to pass, passed
    sysinfo_test.go:91: PASS: Test 2: input 'show.version.3', expected to pass, passed
=== RUN   TestParseShowSystemEnvironmentJsonOutput
--- PASS: TestParseShowSystemEnvironmentJsonOutput (0.00s)
    system_environment_test.go:97: PASS: Test 0: input 'show.environment.1', expected to pass, passed
    system_environment_test.go:97: PASS: Test 1: input 'show.environment.2', expected to pass, passed
=== RUN   TestParseShowSystemResourcesJsonOutput
--- PASS: TestParseShowSystemResourcesJsonOutput (0.00s)
    system_resource_test.go:79: PASS: Test 0: input 'show.system.resources.1', expected to pass, passed
=== RUN   TestParseShowTransceiverJsonOutput
--- PASS: TestParseShowTransceiverJsonOutput (0.00s)
    transceiver_test.go:73: INFO: Test 0: input 'show.interface.transceiver.details.1'
        interface: Ethernet1/1, name: CISCO-AVAGO, serial: ABC1234567U, part_number: 23-4567-89, part_id: AFBR-23ECPZ-CS4, nominal_bit_rate: 25500
        interface: Ethernet1/1, lane: 1, measurements: 44.62,3.25,7.50,0.86,0.19,0
        interface: Ethernet1/1, lane: 2, measurements: 44.62,3.25,7.50,0.86,0.19,0
        interface: Ethernet1/1, lane: 3, measurements: 44.62,3.25,7.50,0.24,0.79,0
        interface: Ethernet1/1, lane: 4, measurements: 44.62,3.25,7.50,0.24,0.79,0
    transceiver_test.go:73: INFO: Test 0: input 'show.interface.transceiver.details.1'
        interface: Ethernet1/36, name: CISCO-AVAGO, serial: ABC1234567U, part_number: 12-3456-78, part_id: AFBR-79EBPZ-CS2, nominal_bit_rate: 25500
        interface: Ethernet1/36, lane: 1, measurements: 36.68,3.26,7.49,-0.37,-0.16,0
        interface: Ethernet1/36, lane: 2, measurements: 36.68,3.26,7.49,-0.32,-0.28,0
        interface: Ethernet1/36, lane: 3, measurements: 36.68,3.26,7.49,0.00,-0.48,0
        interface: Ethernet1/36, lane: 4, measurements: 36.68,3.26,7.49,0.05,-0.30,0
    transceiver_test.go:80: PASS: Test 0: input 'show.interface.transceiver.details.1', expected to pass, passed
    transceiver_test.go:73: INFO: Test 1: input 'show.interface.transceiver.details.2'
        interface: Ethernet1/1, name: CISCO-FINISAR, serial: 8B2C5035A864, part_number: 10-2415-03, part_id: DB2C46ED75C0-C2, nominal_bit_rate: 10300
        interface: Ethernet1/1, lane: 0, measurements: 44.54,3.28,8.67,-1.93,-1.39,0
    transceiver_test.go:80: PASS: Test 1: input 'show.interface.transceiver.details.2', expected to pass, passed
=== RUN   TestParseShowVlanJsonOutput
--- PASS: TestParseShowVlanJsonOutput (0.00s)
    vlan_test.go:81: PASS: Test 0: input 'show.vlans.1', expected to pass, passed
    vlan_test.go:81: PASS: Test 1: input 'show.vlans.2', expected to pass, passed
PASS
coverage: 87.8% of statements
ok  	command-line-arguments	0.382s	coverage: 87.8% of statements
```

**Comment by tao-sj** [comment]
> @tao-sj , could you please draft a few sentences about the dynamics of the configuration method in `README.md`?

good point.

how about this ? 

```
diff --git a/README.md b/README.md
index a258f52..644e088 100644
--- a/README.md
+++ b/README.md
@@ -15,7 +15,8 @@ Cisco NX-OS API client library written in Go.
 
 This library is well documented and comes with a command-line utility for code
 implementation and usage purposes. The library helps its users querying system
-information, vlans, and interfaces, etc.
+information, vlans and interfaces, etc. It also provides API to execute a batch
+of commands, e.g, change interface or vlan configurations.
 
 * `GetSystemInfo()`: **show version**
 * `GetVlans()`: **show vlan**
@@ -27,6 +28,7 @@ information, vlans, and interfaces, etc.
 * `GetBgpSummary()` **show ip bgp summary** (BGP routing summary)
 * `GetTransceivers()` **show interface transceiver details** (fiber transceivers)
 * `GetGeneric()`: runs any arbitrary command and produces JSON output
+* `Configure()`: execute a batch of configuration commands
```

**Comment by greenpau** [comment]
> It also provides API to execute a batch of commands, e.g, change interface or vlan configurations.

@tao-sj , yes, good enough 👍 I think it deserves separate paragraphs:

```
...

Additionally, the library allows "batch" execution of commands.

Moreover, it enables the execution of series of "configure" commands.

For example:
```

File: README.md [hunk 1]
```diff
@@ -28,6 +28,12 @@ information, vlans, and interfaces, etc.
 * `GetTransceivers()` **show interface transceiver details** (fiber transceivers)
 * `GetGeneric()`: runs any arbitrary command and produces JSON output
 
+Additionally, the library allows "batch" execution of configuration commands.
+
```
Comment (greenpau) [inline README.md]: @tao-sj , extra space.

File: README.md [hunk 2]
```diff
@@ -48,6 +54,22 @@ if data, err := cli.GetSystemInfo(); err != nil {
 }
 ```
 
+The following snippet shut down interface e1/1:
```
Comment (greenpau) [inline README.md]: @tao-sj , spot on 💯

Comment (greenpau) [reply README.md]: @tao-sj , minor typo, fix:

"The following snippet shuts down interface e1/1"

File: README.md (hunk 1 above)
Comment (tao-sj) [reply README.md]: sorry, where is it ?

Comment (greenpau) [reply README.md]: @tao-sj , needs `commands, e.g.` (note the period vs. comma)

image

**Comment by greenpau** [comment]
@tao-sj , apparently I can fix it myself in github.

File: README.md (hunk 1 above)
Comment (tao-sj) [reply README.md]: oh, updated.

**Pull request merged**